staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -141,6 +144,7 @@ func newETCD3Client(c storagebackend.TransportConfig) (*clientv3.Client, error)
 		dialOptions = append(dialOptions, grpc.WithContextDialer(dialer))
 	}
 	cfg := clientv3.Config{
+		AutoSyncInterval:     autoSyncInterval,
```

Comment:
I think this isn't backard compatible. There are setups may not want the client to talk to arbitrary etcd instance, but e.g. just to the local one.
I would be fine with giving people ability to set it, but the default should remain 0 as it was.

@jpbetz @lavalamp  - for thoughts

Reply:
@wojtek-t makes sense, any guidance on the best way to attempt to plumb that through? Adding it to the `storagebackend.TransportConfig` that is used as the sole argument to `newETCD3Client` doesn't feel quite right and adding it to `storagebackend.Config` and passing it in separately to `newETCD3Client` also feels a bit off.

Reply:
https://github.com/kubernetes/kubernetes/pull/64746 did this through `storagebackend.Config`. But as @liggitt pointed out, is discussion on that PR about the problems that this can cause.